packages/main/src/DateRangePicker.js

```diff
@@ -0,0 +1,402 @@
+		//
+	},
+};
+
+/**
+ * @class
+ *
+ * <h3 class="comment-api-title">Overview</h3>
+ * The DateRangePicker enables the users to enter a localized date range using touch, mouse, keyboard input, or by selecting a date range in the calendar.
+ *
+ * <h3>Usage</h3>
+ * The user can enter a date by:
+ * Using the calendar that opens in a popup or typing it in directly in the input field (not available for mobile devices).
+ * For the <code>ui5-date-range-picker</code>
+ * <h3>ES6 Module Import</h3>
+ *
+ * <code>import @ui5/webcomponents/dist/DateRangePicker.js";</code>
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.main.DateRangePicker
+ * @extends DatePicker
+ * @tagname ui5-daterange-picker
+ * @since 1.0.0-rc.8
+ * @public
+ */
+class DateRangePicker extends DatePicker {
+	static get metadata() {
+		return metadata;
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get styles() {
+		return [...DatePicker.styles, DateRangePickerCss];
+	}
+
+	static get template() {
+		return DateRangePickerTemplate;
+	}
+
+	static async onDefine() {
+		await DatePicker.define();
+	}
+
+	constructor() {
+		super();
+		this.isFirstDatePick = true;
+	}
+
+	async onAfterRendering() {
+		this.responsivePopover = await this._respPopover();
+		const calendar = this.responsivePopover.querySelector(`#${this._id}-calendar`);
+		const dayPicker = calendar.shadowRoot.querySelector(`#${calendar._id}-daypicker`);
+		dayPicker.addEventListener("item-mouseover", this._itemMouseoverHandler);
+		dayPicker.addEventListener("daypickerrendered", this._keyboardNavigationHandler);
+
+		this._cleanHoveredAttributeFromVisibleItems(dayPicker);
+	}
+
+	_itemMouseoverHandler(event) {
+		const dayItems = event.target.shadowRoot.querySelectorAll(".ui5-dp-item");
+		const firstDateTimestamp = this._selectedDates[0];
+		const lastDateTimestamp = event.detail.target.parentElement.dataset.sapTimestamp;
+
+		for (let i = 0; i < dayItems.length; i++) {
+			if ((dayItems[i].dataset.sapTimestamp < firstDateTimestamp && dayItems[i].dataset.sapTimestamp > lastDateTimestamp)
+				|| (dayItems[i].dataset.sapTimestamp > firstDateTimestamp && dayItems[i].dataset.sapTimestamp < lastDateTimestamp)) {
+				dayItems[i].setAttribute("hovered", "");
+			} else {
+				dayItems[i].removeAttribute("hovered");
+			}
+		}
+	}
+
+	_keyboardNavigationHandler(event) {
+		if (!event.detail.focusedItemIndex) {
+			return;
+		}
+
+		const dayItems = event.target.shadowRoot.querySelectorAll(".ui5-dp-item");
+		const firstDateTimestamp = this._selectedDates[0];
+		const lastDateTimestamp = dayItems[event.detail.focusedItemIndex].dataset.sapTimestamp;
+
+		for (let i = 0; i < dayItems.length; i++) {
+			if ((dayItems[i].dataset.sapTimestamp < firstDateTimestamp && dayItems[i].dataset.sapTimestamp > lastDateTimestamp)
+				|| (dayItems[i].dataset.sapTimestamp > firstDateTimestamp && dayItems[i].dataset.sapTimestamp < lastDateTimestamp)) {
+				dayItems[i].setAttribute("hovered", "");
```

Comment:
Can this setAttrbite/removeAttrbite be done trough the hbs file, in declarative way

Reply:
We discussed this comment offline. We tried to replace the setAttrbite/removeAttrbite  with a field on the day object, representing a single day within the DayPicker, but in order to re-render the days we should update the private _week object on the DayPicker _weeks = [...prevWeeks]. This happens on hover and will rebuild all the weeks. We decided not to do it in this change.